test/index.js

```diff
@@ -23,28 +50,238 @@ test('with metadata', function (t) {
+  run('01', { main: 'contributors-main.js', options: '.' }, ({ file, actual, expected }) => {
     t.is(actual, expected)
+    t.deepEqual(file.messages.map(String), [])
     t.end()
   })
 })
 
 test('with metadata from module (named export)', function (t) {
-  run('01', { main: 'contributors-named.js', contributors: '.' }, ({ cwd, actual, expected }) => {
+  run('01', { main: 'contributors-named.js', options: '.' }, ({ file, actual, expected }) => {
     t.is(actual, expected)
+    t.deepEqual(file.messages.map(String), [])
+    t.end()
+  })
+})
+
+test('with an unfound module id', function (t) {
+  run('00', { options: 'missing.js' }, function ({ err }) {
+    t.ok(/^Error: Cannot find module 'missing.js'/.test(err))
+    t.end()
+  })
+})
+
+test('with a throwing module', function (t) {
+  run('00', { main: 'contributors-throwing.js', options: '.' }, function ({ err }) {
+    t.ok(/^Error: Some error!/.test(err))
+    t.end()
+  })
+})
+
+test('with an invalid exports', function (t) {
+  run('00', { main: 'contributors-invalid-exports.js', options: '.' }, function ({ err }) {
+    t.ok(/^TypeError: The "contributors" option must be \(or resolve to\) an array/.test(err))
+    t.end()
+  })
+})
+
+test('with an invalid contributors setting', function (t) {
+  run('00', { options: { contributors: true } }, function ({ err }) {
+    t.ok(/^TypeError: The "contributors" option must be \(or resolve to\) an array/.test(err))
     t.end()
   })
 })
 
 test('with metadata from module (default export)', function (t) {
-  run('01', { main: 'contributors-default.js', contributors: '.' }, ({ cwd, actual, expected }) => {
+  run('01', { main: 'contributors-default.js', options: '.' }, ({ file, actual, expected }) => {
+    t.is(actual, expected)
+    t.deepEqual(file.messages.map(String), [])
+    t.end()
+  })
+})
+
+test('without heading', function (t) {
+  run('02', { }, ({ file, actual, expected }) => {
+    t.is(actual, expected)
+    t.deepEqual(file.messages.map(String), [])
+    t.end()
+  })
+})
+
+test('without heading, with `appendIfMissing`', function (t) {
+  run('03', { options: { appendIfMissing: true } }, ({ file, actual, expected }) => {
+    t.is(actual, expected)
+    t.deepEqual(
+      file.messages.map(String),
+      ['1:1: no social profile for ' + TEST_EMAIL]
+    )
+    t.end()
+  })
+})
+
+test('with a noreply email', function (t) {
+  const email = '944406+wooorm@users.noreply.github.com'
+  const gitUsers = [[TEST_NAME, email]]
+
+  run('04', { gitUsers }, ({ file, actual, expected }) => {
+    t.is(actual, expected)
+    t.deepEqual(
+      file.messages.map(String),
+      ['1:1: no social profile for ' + email]
+    )
+    t.end()
+  })
+})
+
+test('with a greenkeeper email', function (t) {
+  const email = 'example@greenkeeper.io'
+  const gitUsers = [[TEST_NAME, email]]
+
+  run('00', { gitUsers }, function ({ err }) {
+    t.ok(/^Error: Missing required `contributors` in settings/.test(err))
```

Comment:
So in this case, there's 1 git contributor, filtered out by `remark-git-contributors`, and then `remark-contributors` throws `Missing required contributors` because it gets an empty array. Is that right?

Reply:
Correct, in my words: there’s a Git user, but there are no defined contributors, as this plugin [only includes people that are in Git history as well](https://github.com/remarkjs/remark-git-contributors/blob/1553e47719dfc946101ce1e8a450725abab074ec/index.js#L43), nothing happens.

There are a few cases where this error is thrown (Missing required `contributors`). We could also swallow it of course.

Reply:
Perhaps add a comment here to explain what is tested.

Better yet, rename the test, e.g. `test('ignores greenkeeper email', ..)`